cmd/data.go

```diff
@@ -101,6 +102,33 @@ func (cmd *DataCmd) Execute(ctx context.Context, f *flag.FlagSet, _ ...interface
 	}
 
 	conv := internal.MakeConv()
+
+	client, err := conversion.GetClient(ctx, dbURI)
+	if err != nil {
+		err = fmt.Errorf("can't create client for db %s: %v", dbURI, err)
+		return subcommands.ExitFailure
+	}
+	defer client.Close()
+
+	// If using CSV mode, follow different code path from here.
+	if driverName == constants.CSV {
+		if targetProfile.conn.sp.dbname == "" {
+			err = fmt.Errorf("dbname is mandatory in target-profile for csv source")
+			return subcommands.ExitFailure
+		}
+		// TODO: refactor this to go through DataConv(). Fix it when refactoring passing of source-profile throughout the code
+		// to avoid excess parameters in the DataConv and SchemaConv functions.
+		bw, err := conversion.DataFromCSV(conv, sourceProfile.csv.manifest, client, targetDb)
```

Comment:
Do we check if the schema matches the manifest anywhere in this flow ?

Reply:
Nopes, the assumption is that the spanner schema will match the manifest and has to be taken care of the user. Once we add the querying spanner schema functionality, then we wont need a manifest file at all.